nfts/src/lib.rs

```diff
@@ -239,22 +239,33 @@ pub mod pallet {
 			let who = ensure_signed(origin)?;
 
 			Data::<T>::try_mutate(nft_id, |maybe_data| -> DispatchResult {
-				let data = maybe_data.as_mut().ok_or(Error::<T>::NFTNotFound)?;
-
-				ensure!(data.owner == who, Error::<T>::NotTheNFTOwner);
-				ensure!(!data.listed_for_sale, Error::<T>::CannotDelegateNFTsListedForSale);
-				ensure!(!data.converted_to_capsule, Error::<T>::CannotDelegateCapsules);
-				ensure!(!data.in_transmission, Error::<T>::CannotDelegateNFTsInTransmission);
-
-				if let Some(viewer) = &viewer {
-					ensure!(who != *viewer, Error::<T>::CannotDelegateNFTsToYourself);
+				match maybe_data {
```

Comment:
Instead of having two match arms you can just do in in one line:
"let data = maybe_data.as_mut().ok_or(Error::<T>::NFTNotFound)?;"